### DISCUSSION TITLE
Feature request: Setting default/maximum program volume

### DISCUSSION THREAD

**Discussion by cynr1c** [File-New-Project/EarTrumpet#680] [Ideas]
Hi!
I recently stumbled upon EarTrumpet and I really like it so far. Great job guys!
However I've got a feature request for you: Could you please provide a setting in the options to set the default volume for new programs and/or maximum setting, that can only be surpassed by some special actions (holding ctrl while pulling the slider or something like that)?

The reason for this request is clipping artifacts due to samplerate conversion. I've got an external USB DAC hooked up (Creative SoundBlaster E5), that offers a volume controlled (by Windows main volume) analog output as well as a optionally uncontrolled digital out. I've got a studio DSP/DAC and a seperate volume control connected to that digital out running in direct/pass-through mode. 
So pretty often when a program plays a sound at maximum level with a different samplerate than the DAC is on there is a samplerate conversion taking place which can lead to clipping on high-level sounds. The only viable solution against this is the set each program to -6dB/50% of the system volume, leaving enough headroom for the SRC. Having EarTrumpet auto-set all yet unknown programs to this level would really save a lot of hassle!

**Comment by riverar** [comment]
Understood, we have this in the backlog for our next major release. Stay tuned!

**Reply by vertigo220** [reply]
Fairly certain ET is abandoned at this point. They moved everything out of issues into discussions, which I was concerned about at the time, and it seems I was right, as development seems to have stopped around that time or shortly after (hard to tell, since they don't even bother to put dates for the releases in the change log). This issue, as well as any others users are posting, will likely never be resolved. We need a fork, but there's only one that has any promise after all this time, and it seems very questionable. I think this is something that can be handled through a script, likely powershell, though would probably be clunky. Might be the only option at this point, though.

**Reply by riverar** [reply]
Not abandoned but definitely neglected, my apologies all. My schedule is packed with jobs and life tasks and it's very difficult to find/devote time to OSS maintenance, a very thankless (and non-zero-cost job).

Let me sit down and clean things up. Appreciate the kick in the butt.

**Comment by LarsKemmann** [comment]
I have a different scenario that would benefit from a similar feature. My Bluetooth headset doesn't correctly obey the Windows global system volume, so I need to adjust per-application volume every time I get e.g. a call via Teams or Skype for Business. Having a way to force those applications to always start at 20% volume, for example, would be really helpful for my eardrums. :)

**Comment by sgtdeagle** [comment]
Status so far for this Feature?

**Comment by XJIOP** [comment]
up

**Comment by slbb** [comment]
Similar situation.
When I run a newly installed program, sometimes the default 100% volume is too loud.
If I turn down the global system volume, some programs works well in 100% volume before, these programs' volume will become too low. 

It will be great if there's an option to set a default volume for newly installed program.

**Comment by vertigo220** [comment]
This is by far the biggest feature I need. I've searched without luck for a way to do this, and in fact that's how I came across EarTrumpet, and while it's better than the standards Windows volume control, the fact it lacks this feature was a big disappointment.

The reason I need this is for playing games in Steam. I can't just turn Steam down, since every game launches as its own app, with its own volume, which is always defaulted to 100%, and _every...single...game_ defaults to 747 takeoff level volume that seems to be at least 50% louder, at the same volume, than even an action movie. So what I'd like is the ability to either just have any program EarTrumpet hasn't seen before (can't just be newly shown in EarTrumpet, since it would happen every time an application is restarted) or, better yet, based on the executable path and/or parent process, e.g. all programs launched from C:\Program Files (x86)\Steam or D:\Games or any program launched by steam.exe (though not sure if that would actually work or if that would be the right executable).

And the icing on the cake would be the combined ability, as suggested elsewhere, to be able to control volume with hotkeys, so if by chance a game's volume were actually too low because of this, I could use a hotkey to adjust the volume for all apps in a category, e.g. "Games" (which would be any app that matches the above rule). And then it should remember the volume for any manually changed app/game, while still launching new ones at the default volume. For example, I set "Games" category to 35%, then launch game A and manually increase it to 50%. Later I launch game B, which defaults to 35%, and afterward I relaunch game A, which is at 50%, where I set it before.

Edit: Just had another thought regarding this. I'm not sure if it's possible, but if EarTrumpet can determine the actual volume of an audio stream and adjust it to bring it to a specified level, that would possibly be an even better solution. For example, if I could just set any new app to default to a volume level that corresponds to 90 dB, so one app launches and is really loud (e.g. 110 dB), it would be turned down appropriately, and vice-versa for a quieter app.

**Comment by ghost** [comment]
This would fix all the games that scream at you the first time you open them. I'd really love to see this espc. after riverar wrote it is coming in the next major release 18 months ago

**Comment by LukasDoubleU** [comment]
I implemented this and other features over at my fork.
There is no max volume for single apps, but you can configure a default volume that will be applied to all new apps.

If you want you can test the pre-release:
https://github.com/LukasDoubleU/EarTrumpet/releases

**Comment by vertigo220** [comment]
I see a bunch of issue were just locked and moved to discussion. Why was this done?

**Comment by vertigo220** [comment]
@LukasDoubleU - there's no issues page for your repo, so testers can't easily report problems

**Reply by LukasDoubleU** [reply]
there are some developments going on that make my changes obsolete, thus I won't continue to work on them.

**Reply by vertigo220** [reply]
Why then did you post suggesting people test it? And why did _tons_ of issues just get closed? What am I missing?

**Reply by riverar** [reply]
We moved all feature requests to the Discussions area, keeping the Issues area for actual issues.

**Reply by vertigo220** [reply]
Thanks! I just wish a comment had been made about that when closing them, as well as telling us we would remain subscribed to discussions from issues we were subscribed to (I had to check for myself since I didn't know if it would transfer over). As it was done, it looked like they were just killing them all. Got worried with the flood of emails. :)

**Reply by riverar** [reply]
It does say that at the bottom of each closed issue, but I don't think it shows up on mobile. 😓 I'll create a top-level issue to maybe help clarify, thanks!

**Reply by vertigo220** [reply]
That's definitely part of the problem, but even when I came to the site to check it out and saw the message, it didn't say _why_ they were moved to discussion, so I wasn't sure what was going on. It looked like somebody decided they weren't going to bother with any of them and so just closed them, but left them open for further discussion anyways. Was just unclear and would have made a lot more sense if there had been a quick comment copy/pasted to each of them before closing explaining what was happening.

**Comment by RedSnt** [comment]
I feel like this should be able to accomplish using actions (dev build), but I'm not certain how it's done. I tried messing around, but didn't figure it out on my own, so does anyone have the magic to make this work?  Or is it still a "planned feature" 4 years in the making?
